src/reanimated2/PlatformChecker.ts

```diff
@@ -32,8 +32,17 @@ export function nativeShouldBeMock() {
   return isJest() || isChromeDebugger() || isWindows();
 }
 
+export function isWindowAvailable() {
+  // the window object is unavailable when building the server portion of a site that uses SSG
+  // this function shouldn't be used to conditionally render components
+  // https://www.joshwcomeau.com/react/the-perils-of-rehydration/
+  return typeof window !== 'undefined';
```

Comment:
Why not just
```suggestion
  return !!window;
```

Reply:
In this version the server build still fails for me :(

Reply:
```
KNWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMNXXXX
K0NMMMMMMMWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMXOkOO
0OXMMMMWNK0XWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMNXOx
OkKWMMWX00O0NMMMMMMMWXKNMMMMMMMMMMMMMMMMMMMMMMMMMMWWMMMMMWKx
O00KXXKO00O0WMMMMMMXO0xokXWWMMMMMMMMMMMMMMMMMMMMNKKKXXWMMWKk
KXXX0kkOO0XWMMMMMMNdlOklccloxkO0KKNWMMMMMMMMMMMMXOOKKKKWMWKO
0KKKOkk0NWMMMMMMW0olx0KKKOdlcldxkkkOKXWMMMMMMMMMWX00KK0XWXOk
OOOOOO0NMMMMMMMM0lo0XXXXKKKKKKKKK0Okkk0NMMMMMMMMMWKO00O0K0OO
O00K0OXMMMMMWNKkooOKKK0kxkkkkkOO000OOkkOXWMMMMMMMMXOO0Ok0K0O
00KK00XMMWNK0OxdxxkxxkOkxO00OkkkxxkOOOkkOKNMMMMMMMN0O0Ok0KK0
0KKK00XWX0kxxxxkO0OkkOOkxxkkO000OkxkOOOkkkOXWMMMMMN0O0000000
0KK000XKxxxkOO00Okdolc:;;;:coxkO00OkOOOOOkkk0XWMMMXOO00OO000
kO00OOOkxxO000Okdc,.........',;:dkOOOOOOOOOOkO0NMMXOOOkk000O
OkkkkkkkOOOOOOko,...............':dOOOOOOOOOOkk0NMXOOkkOKK0O
K0kxxkOOOOOOOOo,..................,lkOO0OOOOOOkkKWNOkxxOKK00
O0OkkOOOOOOOko,........''..........'ckOOOOOOOOkk0NWXOddOKK0O
XK0OOOOOOOkl;'.....':ldkkxoc,.......'ckOOOOOOOOOOKWMWKOO000O
MWKO0OOOOx:......,lxO0KKKKK0x:'......':xOOOOOOOOO0NMMMNKOOO0
MN0O0OOOd:.....'cxO000KKKK000kdc'......:xOOOOOOO0OXMMMMWNNNW
MW0OOOOx:...',cdO00000000000000Od:'.....;dOOOOOOO0NMMMMMMMMM
MMX0kOkc'.':oxOOO0000000000000000Od:.....;dkOOkOKWMMMMMMMMMM
MMMNOkd:';oxOOOOOOO00O00000OOOO00OOkl,....;okkOXWMMMMMMMMMMM
MMMMXOkdodkkOOOOOOOOOOOOOOOOOOOOOOOOOdc;,,:oxOXWMMMMMMMMMMMM
MMMMWXKOkkkkkkkkkkkkkkkkkkkkkkkkO00000kxdodO0XWMMMMMMMMMMMMM
```